### DISCUSSION TITLE
Feature request: After closing a window, focus the most recent window

### DISCUSSION THREAD

**Discussion by ghost** [valinet/ExplorerPatcher#2154] [Ideas]
In Windows 10 and 11 when you switch windows by clicking their Taskbar button (i.e., not by Alt+Tab), and then close them, focus goes to the Taskbar and not to the most recent window. Example:

- Open an app, e.g. Windows Explorer.
- Open a different app, e.g. Edge.
- Now the second app has focus, thus focus the first _by clicking its Taskbar button_, and close it.
- **Currently**: The Taskbar gets focused. **My expectation**: The most recent window (Edge) should get focused.

I think, it would be nice, if ExplorerPatcher would fix this behavior.

Oftentimes I don't immediately notice, that the window I'm working with has no focus, because in latest Windows you can scroll a window with the mouse wheel even if it's not focused. Thus I can read a long article in my browser that's not focused, and when I press a hotkey, the browser naturally doesn't react. Very annoying.

**Comment by pyrates999** [comment]
Do you mean this?

1. Close all other windows
2. Open an application, e.g. Windows Explorer.  **If it opens maximized, click the maximize button to unmaximize it, close it and start over.**
3. Open a different application, e.g. Edge.  **If it opens maximized, click the maximize button to unmaximize it, close it and start over.**
4. Now the second application has focus.
5. Right click on the first app in the taskbar and select close window.

Currently: The Taskbar gets focused.  If you have another application that is already open, it may get focus instead of the application opened in step 3.

What should happen: The most recent window (Edge) should get focused.

-------

Please verify this without EP installed.  If it happens without EP installed, this is a bug in windows and thus EP can't fix it.

**Reply by Poopooracoocoo** [reply]
As I said in one of the other issues or discussions, I saw this behaviour on Windows 10 without EP. I think EP could perhaps implement a change for this. I don't know about the technical feasibility, but it's not necessarily out of scope as far as I can tell :)

**Reply by pyrates999** [reply]
I would leave this for MS to fix.

**Reply by ghost** [reply]
The bug is present in windows for years now. I think they simply don't care. They won't fix it.
